tests/test_config_user_groups.cpp

```diff
@@ -0,0 +1,369 @@
+    CHECK(room4 == "someroom");
+    CHECK(room5 == "someroom");
+    CHECK(room6 == "someroom");
+    CHECK(room7 == "someroom");
+    CHECK(room8 == "someroom");
+    CHECK(community::canonical_room(room1) == "someroom");
+    CHECK(community::canonical_room(room2) == "someroom");
+    CHECK(community::canonical_room(room3) == "someroom");
+    CHECK(community::canonical_room(room4) == "someroom");
+    CHECK(community::canonical_room(room5) == "someroom");
+    CHECK(community::canonical_room(room6) == "someroom");
+    CHECK(community::canonical_room(room7) == "someroom");
+    CHECK(community::canonical_room(room8) == "someroom");
+    CHECK(to_hex(pk1) == "0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef");
+    CHECK(to_hex(pk2) == "0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef");
+    CHECK(to_hex(pk3) == "0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef");
+    CHECK(to_hex(pk4) == "0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef");
+    CHECK(to_hex(pk5) == "0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef");
+    CHECK(to_hex(pk6) == "0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef");
+    CHECK(to_hex(pk7) == "0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef");
+    CHECK(to_hex(pk8) == "0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef");
+}
+
+TEST_CASE("User Groups", "[config][groups]") {
+
+    const auto seed = "0123456789abcdef0123456789abcdef00000000000000000000000000000000"_hexbytes;
+    std::array<unsigned char, 32> ed_pk, curve_pk;
+    std::array<unsigned char, 64> ed_sk;
+    crypto_sign_ed25519_seed_keypair(
+            ed_pk.data(), ed_sk.data(), reinterpret_cast<const unsigned char*>(seed.data()));
+    int rc = crypto_sign_ed25519_pk_to_curve25519(curve_pk.data(), ed_pk.data());
+    REQUIRE(rc == 0);
+
+    REQUIRE(oxenc::to_hex(ed_pk.begin(), ed_pk.end()) ==
+            "4cb76fdc6d32278e3f83dbf608360ecc6b65727934b85d2fb86862ff98c46ab7");
+    REQUIRE(oxenc::to_hex(curve_pk.begin(), curve_pk.end()) ==
+            "d2ad010eeb72d72e561d9de7bd7b6989af77dcabffa03a5111a6c859ae5c3a72");
+    CHECK(oxenc::to_hex(seed.begin(), seed.end()) ==
+          oxenc::to_hex(ed_sk.begin(), ed_sk.begin() + 32));
+
+    session::config::UserGroups groups{ustring_view{seed}, std::nullopt};
+
+    constexpr auto definitely_real_id =
+            "055000000000000000000000000000000000000000000000000000000000000000"sv;
+
+    CHECK_FALSE(groups.get_legacy_group(definitely_real_id));
+
+    CHECK(groups.empty());
+    CHECK(groups.size() == 0);
+
+    auto c = groups.get_or_construct_legacy_group(definitely_real_id);
+
+    CHECK(c.session_id == definitely_real_id);
+    CHECK_FALSE(c.hidden);
+    CHECK(c.disappearing_timer == 0min);
+    CHECK(c.enc_pubkey.empty());
+    CHECK(c.enc_seckey.empty());
+    CHECK(c.priority == 0);
+    CHECK(c.name == "");
+    CHECK(c.members().empty());
```

Comment:
Is there a C interface for the user groups legacy group `members`?

Reply:
Will open a new PR with the additions.